packages/cli-kit/src/public/node/git.ts

```diff
@@ -408,18 +382,39 @@ export async function removeGitRemote(directory: string, remoteName = 'origin'):
   outputDebug(outputContent`Removing git remote ${remoteName} from ${outputToken.path(directory)}...`)
   await ensureGitIsPresentOrAbort()
 
-  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-  // @ts-ignore
-  const repo = git(directory)
+  await withGit({directory}, async (repo) => {
+    // Check if remote exists first
+    const remotes = await repo.getRemotes()
+    const remoteExists = remotes.some((remote: {name: string}) => remote.name === remoteName)
+
+    if (!remoteExists) {
+      outputDebug(outputContent`Remote ${remoteName} does not exist, no action needed`)
+      return
+    }
 
-  // Check if remote exists first
-  const remotes = await repo.getRemotes()
-  const remoteExists = remotes.some((remote: {name: string}) => remote.name === remoteName)
+    await repo.removeRemote(remoteName)
+  })
+}
 
-  if (!remoteExists) {
-    outputDebug(outputContent`Remote ${remoteName} does not exist, no action needed`)
-    return
+async function withGit<T>(
+  {
+    directory,
+  }: {
+    directory?: string
+  },
+  callback: (git: SimpleGit) => Promise<T>,
+): Promise<T> {
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-ignore
+  const repo = git({baseDir: directory})
+  try {
+    return await callback(repo)
+  } catch (err) {
+    if (err instanceof Error) {
```

Comment:
Will the caught error ever _not_ be an instance of `Error`? I assumed `Error` is the base error class every other error inherits from.